src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -583,92 +372,44 @@ private Future<ReceiptStatus> calculatePoLineReceiptStatus(List<Piece> byPoLine,
     return Future.succeededFuture(receivedQty == 0 ? AWAITING_RECEIPT : PARTIALLY_RECEIVED);
   }
 
-  private Future<List<Piece>> getPiecesByPoLine(String poLineId, RequestContext requestContext) {
-    String query = String.format("poLineId==%s", poLineId);
-    return pieceStorageService.getPieces(Integer.MAX_VALUE, 0, query, requestContext)
-      .map(PieceCollection::getPieces);
-  }
-
-  /**
-   * Checks if there is processing error for particular piece record
-   * @param piece piece record to get error for
-   * @return {@code true} if there is an error associated with piece record
+  //-------------------------------------------------------------------------------------
+  /*
+  filterMissingLocations
    */
-  private boolean isSuccessfullyProcessedPiece(Piece piece) {
-    return getError(piece.getPoLineId(), piece.getId()) == null;
-  }
-
-  /**
-   * Gets processing error for particular piece record
-   * @param polId PO Line id
-   * @param pieceId piece id
-   * @return error object if presents or null
-   */
-  protected Error getError(String polId, String pieceId) {
-    return Optional.ofNullable(processingErrors.get(polId))
-     .map(errors -> errors.get(pieceId))
-     .orElse(null);
-  }
-
-  void addError(String polId, String pieceId, Error error) {
-    processingErrors.computeIfAbsent(polId, k -> new HashMap<>())
-      .put(pieceId, error);
-  }
-
-  public void calculateProcessingErrors(String poLineId, ReceivingResult result,
-      Map<String, Piece> processedPiecesForPoLine, Map<String, Integer> resultCounts, String pieceId) {
-    // Calculate processing status
-    ProcessingStatus status = new ProcessingStatus();
-    Error error = getError(poLineId, pieceId);
-    if (processedPiecesForPoLine.get(pieceId) != null && error == null) {
-      status.setType(ProcessingStatus.Type.SUCCESS);
-      resultCounts.merge(ProcessingStatus.Type.SUCCESS.toString(), 1, Integer::sum);
-    } else {
-      status.setType(ProcessingStatus.Type.FAILURE);
-      status.setError(error);
-      resultCounts.merge(ProcessingStatus.Type.FAILURE.toString(), 1, Integer::sum);
-    }
-
-    ReceivingItemResult itemResult = new ReceivingItemResult();
-    itemResult.setPieceId(pieceId);
-    itemResult.setProcessingStatus(status);
-    result.getReceivingItemResults().add(itemResult);
-  }
 
   /**
    * Filter by locationId presence for items/pieces related to POLine.
    *
    * @return {@link Future} which holds map with PO line id as key
-   *         and list of corresponding pieces as value
+   * and list of corresponding pieces as value
    */
   Future<Map<String, List<Piece>>> filterMissingLocations(Map<String, List<Piece>> piecesRecords, RequestContext requestContext) {
     return getPoLines(StreamEx.ofKeys(piecesRecords).toList(), requestContext)
       .map(poLines -> {
-        for(PoLine poLine : poLines) {
+        for (PoLine poLine : poLines) {
```

Comment:
we can use poLines.forEach

Reply:
From old code as well. Does forEach has any advantage when block inside is nontrivial?

Reply:
code will be less, and lead to easier to understand. if it is not suppose to refactor, I will resolve comment